### PR TITLE
AlmaLinux/build-system#24: Add mechanism of generating community PGP keys; AlmaLinux/build-system#28: Fix sign node process freeze after new sign key was generated

### PR DESCRIPTION
**Pull request by soksanichenko**
- The subject is implemented
- Some refactoring

### CODE REVIEW

**Review by javihernandez — COMMENTED** [review]
Looks good on paper :sweat_smile: 
Can this be deployed on staging so we can give it a try?
Also, I see that we have a mix of single and double quotes, do you mind trying to homogenize it?

File: sign_node/uploaders/pulp.py [hunk 1]
```diff
@@ -198,28 +203,32 @@ def upload(self, artifacts_dir: str) -> List[str]:
...
         ----------
-        artifacts_dir : str
+        filename : str
             Path to files that need to be uploaded.
+        artifact_type : str or None
+            Type of uploaded artifact
 
         Returns
         -------
         list
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply sign_node/uploaders/pulp.py]: Fixed

File: sign_node/uploaders/pulp.py [hunk 2]
```diff
@@ -198,28 +203,32 @@ def upload(self, artifacts_dir: str) -> List[str]:
...
+            self, filename: str,
+            artifact_type: typing.Optional[str] = None,
+    ) -> Artifact:
         """
 
         Parameters
         ----------
-        artifacts_dir : str
+        filename : str
             Path to files that need to be uploaded.
```
[the remark this answers is not in the document]
Comment (soksanichenko) [reply sign_node/uploaders/pulp.py]: Fixed

**Comment by soksanichenko** [comment]
The latest changes are already deployed to the first staging. But you can try it if can push data directly to the websocket, becase the endpoint from side of web-server I'm still writing.
The quotes are fixed.

**Comment by soksanichenko** [comment]
Closed in favor to https://github.com/AlmaLinux/albs-sign-node/pull/22

**Pull request closed without merging**